edx/edx-ora2
# fix: submission button take too long to show up

When the user save the answer on tinymce then reload the page, the submission button can take a really long time to show up.

[AU-6]

**TL;DR -** `tinymce` instantiation was asynchronous and it miss the opportunity to give response.

JIRA: [AU-6](https://openedx.atlassian.net/browse/AU-6)

**What changed?**

- Wait for `tinymce` to finish initialize  before moving forward
- It should have failed but the process seem to get run again from different method (not sure if it is intended to be a fail safe or it was even intended at all). Did not have to time to investigate on minified code. But the behavior was definitely different from normal `textarea`.

**Developer Checklist**


**Testing Instructions**

- Create the bug
  -  Create an ORA with WYSIWYG editor
  - Input something
  - Refresh the page. 
 **Note**: this bug can be visually inconsistence on the local machine but it is more consistence on [stage](https://learning.stage.edx.org/course/course-v1:edx+TS101+2020_T1/block-v1:edx+TS101+2020_T1+type@sequential+block@8eb9943459334f75940dd46a6e05b7d6/block-v1:edx+TS101+2020_T1+type@vertical+block@3a4dcdcffb6e4405ba324a482b79b8b7)
- To see the change
  - add `console.log(textFieldsIsNotBlank)` to [this line](https://github.com/edx/edx-ora2/blob/70e2c02cf1d36ac0af952ff91a878166f29c4bf0/openassessment/xblock/static/js/src/lms/oa_response.js#L209)
  - then use `make static` and copy generated the `javascript`'s change
  - copy `openassessment/xblock/static/dist/openassessment-lms.js` and `openassessment/xblock/static/dist/openassessment-editor-tinymce.jss` to `openassessment/xblock/static/js/`. (webpack issue)
  - Refresh the bug page, you should see log `false`
  - **REASON**: [this.editorInstances](https://github.com/edx/edx-ora2/blob/70e2c02cf1d36ac0af952ff91a878166f29c4bf0/openassessment/xblock/static/js/src/lms/editors/oa_editor_tinymce.js#L118) is empty because `tinymce` hasn't been [instantiated](https://github.com/edx/edx-ora2/blob/70e2c02cf1d36ac0af952ff91a878166f29c4bf0/openassessment/xblock/static/js/src/lms/editors/oa_editor_tinymce.js#L50) in the call back yet.
  - The change was to make `tinymce` instantiation more synchronous.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:


FYI: @edx/masters-devs-gta
